docs/architecture.md

```diff
@@ -112,71 +112,199 @@ The redux store is composed of two `slices` : `documents` and `queries`:
 
-Our two slices `documents` and `queries` listen actions and do some specific work on
- `isReceivingMutationResult` action.
+#### Auto query update
 
-`documents`: If the `_id` of the mutated document is not present, then we add the document. 
-If the `_id` is already there, then we update the content of the document with the fresh data
-(for now the work is done in extractAndMergeDocument() method). 
+Since there is a link between queries and documents through the ids list in each query object, we need to be able to deal with changes occuring in the documents collection, i.e. when documents are added or deleted.
 
-So if your app is linked to the documents store via getDocumentFromStore() for instance
-your app will have the updated value. 
+For this, cozy-client has a mechanism called "query updater": each time a query is run, all the queries in the store are re-evaluated directly in the store.
+That means that for each query, we take its definition, convert it into a mango predicate, and run it against the store's document collection.
+For each query result, we check if there is any change in the known ids list, and update it accordingly.
 
-`queries` has an autoUpdater mechanism that does something we can explain this way:
-- it takes the mutated documents (newly created or updated) 
-- it converts the initial `query` to a "`js predicate`" (thanks to the [sift library](https://github.com/crcn/sift.js/))
-- For each query already in the `slice` it runs this `js predicate` and detects if the query
-is concerned by the mutation
-- If the query is concerned, then it checks if it has to remove / add the id of the `mutated` 
-document
-(for now the work is done mainly in queries.js/updateData())
+Thanks to this, we ensure that all the queries are always up-to-date, and apps can be effectively rendered thanks to the [react integration](/docs/react-integration.md).
 
-So in our previous example our `todoslist` is concerned by the addition of the new todo, then 
-the `id` is added to `todolist` data, then the component linked will be refreshed with this new 
-document.
 
-### About updates
+#### Fetch policy and query naming
 
-As already mentioned, when a mutated document already exists in the store throught its `_id`, it is updated with the new content. However, this update never removes existing fields, it only merges or adds new fields. This is typically useful when two queries retrieve the same document, but not with all the fields. 
+The fetch policy is a simple mechanism useful to prevent excessive fetching. 
+It is optionnaly defined for each query and define how many time should be elapsed since the last query execution before running again the query.
+
+In this example we query the checked todos, only if the last execution occured more than 10 minutes ago:
+```js
+const queryDef = Q('io.cozy.todos').where(checked: true)
+const queryOptions = {
+  as: 'checkedTodos', // query unique name, to find it in the store
```

Comment:
We could follow the best practice which I proposed to introduce in a previous comment before. This would give `io.cozy.todos/checked/true`